lib/devtools/rake/flay.rb

```diff
@@ -0,0 +1,110 @@
+module Devtools
+  module Rake
+    # Flay metric runner
+    class Flay
+      include Anima.new(:threshold, :total_score, :lib_dirs, :excludes), Procto.call(:verify), Adamantium
+
+      BELOW_THRESHOLD = 'Adjust flay threshold down to %d'.freeze
+      TOTAL_MISMATCH  = 'Flay total is now %d, but expected %d'.freeze
+      ABOVE_THRESHOLD = '%d chunks have a duplicate mass > %d'.freeze
+
+      # Verify code specified by `files` does not violate flay expectations
+      #
+      # @raise [SystemExit] if a violation is found
+      # @return [undefined] otherwise
+      #
+      # @api private
+      def verify
+        # Run flay first to ensure the max mass matches the threshold
+        Devtools.notify_metric_violation(BELOW_THRESHOLD % largest_mass) if below_threshold?
+        Devtools.notify_metric_violation(TOTAL_MISMATCH % [total_mass, total_score]) if total_mismatch?
+
+        # Run flay a second time with the threshold set
+        return unless above_threshold?
+
+        restricted_flay_scale.flay_report
+        Devtools.notify_metric_violation(ABOVE_THRESHOLD % [restricted_mass_size, threshold])
+      end
+
+      private
```

Comment:
We typically place the private in the same indentation as the opening `class` / `module` keyword.